corehq/util/jsonattrs.py

```diff
@@ -100,6 +101,8 @@ def __init__(self, *args, builder, **kw):
         self.builder = builder
 
     def get_prep_value(self, value):
+        if isinstance(value, Expression):
```

Comment:
Given [this line](https://github.com/django/django/blob/0ff46591ac3010841c73fd26e0fef93995fedd99/django/db/models/sql/compiler.py#L1618) just above the [code that you linked](https://github.com/django/django/commit/0ff46591ac3010841c73fd26e0fef93995fedd99#diff-f58de2deaccecd2d53199c5ca29e3e1050ec2adb80fb057cdfc0b4e6accdf14fL1640), I think we should use `hasattr(value, "resolve_expression")` here. [`resolve_expression` is an attribute of `BaseExpression`](https://github.com/django/django/blob/0ff46591ac3010841c73fd26e0fef93995fedd99/django/db/models/expressions.py#L258), which suggests there may be expression types that do not inherit from `Expression`.

Reply:
See 379cf00f21d14b34e2a46a9ce720bde6afb7f91b